lib/Model.php

```diff
@@ -1667,7 +1667,8 @@ public static function find_by_pk($values, $options)
 		}
 		$results = count($list);
 
-		if ($results != ($expected = count($values)))
+		$count_values = (!is_array($values)) ? array($values) : $values;
```

Comment:
@koenpunt Per your request I changed the check to use empty() in SQLBuilder. Here in Model::find_by_pk() you can not do that, as $values can be either a single primary key, or an array of primary keys. Meaning it's not empty but it's also not always an array. If you then try to count it, this is what you get:

> 1) MysqlAdapterTest::test_query_column_info
count(): Parameter must be an array or an object that implements Countable

The whole point of my PR is to avoid triggering these kind of errors, as they pollute error logs and error notification scripts. In some enterprise applications this literally results in hundreds of mail reports a day. So I'd advise not using empty() here as it generates warnings in PHP and even errors in PHPUnit.

Anyway, if you'd try to run this code yourself with empty() you'll see what I mean.